AdminWebsite/AdminWebsite/Startup.cs

```diff
@@ -68,12 +71,17 @@ private void RegisterAuth(IServiceCollection serviceCollection)
             {
                 options.DefaultAuthenticateScheme = JwtBearerDefaults.AuthenticationScheme;
                 options.DefaultChallengeScheme = JwtBearerDefaults.AuthenticationScheme;
+
             }).AddJwtBearer(options =>
             {
                 options.Authority = securitySettings.Authority;
                 options.TokenValidationParameters.ValidateLifetime = true;
                 options.Audience = securitySettings.ClientId;
                 options.TokenValidationParameters.ClockSkew = TimeSpan.Zero;
+                options.Events = new JwtBearerEvents
```

Comment:
You could even create your own class here, subclassing JwtBearerEvents and including a method for the OnTokenValidated handler. That way you can discreetly tuck away the functionality somewhere not in Startup.cs.

Reply:
Yeh I did think about that but no point at the moment as its only ever used here and not reusable ever probably. Can push it out if the need arises ?